drivers: gpio: pca series: missing dereference in pointer assignment

**Describe the bug**

When `CONFIG_GPIO_PCA_SERIES_INTERRUPT=n`, a `uint32_t` value was being assigned to a `gpio_port_value_t *` in the driver rather than assigning the target o the `gpio_port_value_t *`.

This had an effect other than the one intended but was likely just a typo.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? Yes. e79507b5e372640bec86929ed4322259fc088b59

**To Reproduce**

Steps to reproduce the behavior:
1. create the board overlay files at the bottom of this issue
2. west build -b nucleo_g071rb tests/drivers/gpio/gpio_api_1pin
4. See warning (promoted to error with `twister`)

**Expected behavior**
The driver should compile without warnings or errors.

**Impact**
It can cause a bus fault when the `LOG_DBG()` statement is executed. Also, generally, will not read back the correct data from the port.

**Logs and console output**
```
/home/cfriedt/tt-zephyr-platforms-work/zephyr/drivers/gpio/gpio_pca_series.c: In function 'gpio_pca_series_port_read_standard':
/home/cfriedt/tt-zephyr-platforms-work/zephyr/drivers/gpio/gpio_pca_series.c:1071:23: warning: assignment to 'gpio_port_value_t *' {aka 'unsigned int *'} from 'uint32_t' {aka 'unsigned int'} makes pointer from integer without a cast [-Wint-conversion]
 1071 |                 value = sys_le32_to_cpu(input_data);
      |                       ^
```

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK v0.17.0
 - Commit SHA or Version used: c35bb0de8023750de5ba31dcfbd7b5277466147e

**Additional context**
`tests/drivers/gpio/gpio_api_1pin/boards/nucleo_g071rb.conf`
```
CONFIG_I2C=y
CONFIG_GPIO_PCA_SERIES_INTERRUPT=n
```

`tests/drivers/gpio/gpio_api_1pin/boards/nucleo_g071rb.overlay`
```
/ {
	aliases {
		led0 = &expansion_led_0;
	};
};

&i2c1 {
    status = "okay";

    gpiox0: pca9554a@38 {
		compatible = "nxp,pca9554";
		reg = <0x38>;
		ngpios = <8>;
		int-gpios = <&gpioa 8 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
        status = "okay";

		gpio-controller;
		#gpio-cells = <2>;
	};
};

&leds {
	expansion_led_0: led_0 {
		gpios = <&gpiox0 0x0 0x0>;
		label = "Expansion LED";
	};
};
```

--- a/drivers/gpio/gpio_pca_series.c
+++ b/drivers/gpio/gpio_pca_series.c
@@ -1068,7 +1068,7 @@ static int gpio_pca_series_port_read_standard(
 	if (ret) {
 		LOG_ERR("port read error %d", ret);
 	} else {
-		value = sys_le32_to_cpu(input_data);
+		*value = (gpio_port_value_t)sys_le32_to_cpu(input_data);
 	}
 	k_sem_give(&data->lock);
 #endif /* CONFIG_GPIO_PCA_SERIES_INTERRUPT */

